scratch/camera_feed/display_camera_feed.cpp

```diff
@@ -0,0 +1,165 @@
+#define CAMERA_FRAME_BUFFER_COUNT 5
+#define WINDOW_X_LENGTH 1920 //window resolution. Standard is 1920
+#define WINDOW_Y_LENGTH 1080 //my window resolution. Standard is 1080
+
+const long unsigned int _jpegSize = WINDOW_X_LENGTH * WINDOW_Y_LENGTH * 3;
+
+void decompress(unsigned char* buffer, unsigned char* compressed_image, int pitch){
+    tjhandle _jpegDecompressor = tjInitDecompress();
+    if(_jpegDecompressor == NULL)
+        std::cout << "decompressor was not made" << std::endl;
+    /*
+        The parameters are the decompressor, the compressed image, the size of the
+        image as an unsigned long, the buffer, the width of the jpeg, the pitch,
+        the height, the pixel type (We have RGB here each contained in 1 byte),
+        and the flags. Basically we say decompress it quickly.
+    */
+    //std::cout << ". JPG size: " << _jpegSize <<  ". Pitch " << pitch << std::endl;
+    if (tjDecompress2(_jpegDecompressor, compressed_image, _jpegSize, buffer, WINDOW_X_LENGTH, pitch, WINDOW_Y_LENGTH, TJPF_RGB, TJFLAG_FASTDCT) == -1)
+        std::cout << "decompressed incorrectly" << std::endl;
+    //free the decompressor.
+    tjDestroy(_jpegDecompressor);
+}
+
+int main(){
+    int pitch = 1920*3;
+    uint32_t offset;
+    //create a buffer for the message struct (declared in loop)
+    unsigned char* temp_compressed_message_buffer = new unsigned char[CAMERA_FRAME_BUFFER_SIZE];
+    //array of buffers to hold the compressed messages
+    BufferItem* frame_buffer = new BufferItem[CAMERA_FRAME_BUFFER_COUNT];
+    //pointer to where we keep the pixel data
+    unsigned char* pixel_location;
+    for(int i = 0; i < CAMERA_FRAME_BUFFER_COUNT; i++){
+        frame_buffer[i].data_sections = new unsigned char[CAMERA_FRAME_BUFFER_SIZE];
+    }
+    uint8_t indx = 0;
+    network::Connection conn;
+    const char* destination_port = "127.0.0.1";
+    //allow error to fade out of scope
+    {
+        network::Error error = network::connect(&conn, destination_port, 45546, 45545) ;
+        if(error == network::Error::OPEN_SOCKET)
+            std::cout << "Open socket error: " << std::endl;
+        else if(error == network::Error::BIND_SOCKET)
+            std::cout << "Binding socket error" << std::endl;
+    }
+    //check for failure
+    if (SDL_Init(SDL_INIT_VIDEO) != 0){
+        std::cout << "SDL_Init Error: " << SDL_GetError() << std::endl;
+        return 1;
+    }
+    //creates window. pramaeters are window title, x start pos, y start pos,
+    //x end pos, and y end pos. We use the dimensions of the window provided above
+    SDL_Window *win = SDL_CreateWindow("Camera feed!", 0, 0, WINDOW_X_LENGTH, WINDOW_Y_LENGTH, SDL_WINDOW_SHOWN);
+    //check for failure
+    if (win == nullptr){
+        std::cout << "SDL_CreateWindow Error: " << SDL_GetError() << std::endl;
+        SDL_Quit();
+        return 1;
+    }
+    /*
+        makes a renderer. args are the window to render to, the index of the 
```

Comment:
Make these comments somewhat proper sentences. Don't capitalize variable names, but try to have full capitalized sentences; it just makes the comments look more professional / clean.